src/machine.rs

```diff
@@ -249,6 +249,22 @@ impl<'m> Machine<'m> {
         Ok(())
     }
 
+    /// Clean up resources allocated for machine.
+    #[instrument(skip_all)]
+    pub async fn delete(&self) -> Result<(), Error> {
```

Comment:
* This should take `self` by value, so the Machine isn't used after this call.
* This should either shutdown/stop the machine first or we verify that machine is stopped before we just delete the resources from under it?

Reply:
Self by value - sounds good

Reply:
Agree we should not remove resources from under running machine. Imho it will be easier for users if we provide a single method for completely getting rid of some machine. Meaning we should
1) shutdown
2) delete if not running

Reply:
right, care needs to be taken to not have error from `shutdown` propagate to the caller of `delete`, in case machine has already been shut down.